src/storage/txn/actions/check_data_constraint.rs

```diff
@@ -32,7 +32,11 @@ pub(crate) fn check_data_constraint<S: Snapshot>(
     // 2.The current write type is `Rollback` or `Lock`, and the key have an older
     // version.
     if write.write_type == WriteType::Put || reader.key_exist(key, write_commit_ts.prev())? {
-        return Err(ErrorInner::AlreadyExist { key: key.to_raw()? }.into());
+        return Err(ErrorInner::AlreadyExist {
+            key: key.to_raw()?,
+            existing_start_ts: write.start_ts,
```

Comment:
If the previous existed version is found in the `reader.key_exist` branch, `write` should be the latest `Lock` or `Rollback` record, and it's start_ts doesn't match the meaning of the `existing_start_ts` field.

Reply:
The logic is fixed, PTAL